src/mlpack/methods/kernel_svm/kernel_svm_impl.hpp

```diff
@@ -0,0 +1,287 @@
+        }
+        else
+        {
+          L = std::max(0.0, alpha(j) - alpha(i));
+          H = std::min(regularization, regularization +
+                       alpha(j) - alpha(i));
+        }
+
+        if (L == H)
+          continue;
+
+        // Compute eta by (14).
+        eta = 2 * K(i, j) - K(i, i) - K(j, j);
+        if (eta >= 0)
+          continue;
+        // Compute and clip new value for alpha j using (12) and (15).
+        alpha(j) = alpha(j) - (trainLabels(j) * (E(i) - E(j))) / eta;
+
+        // Clip.
+        alpha(j) = std::min(H, alpha(j));
+        alpha(j) = std::max(L, alpha(j));
+
+        // Check if change in alpha is noticeable or not.
+        if (std::abs(alpha(j) - alpha_j_old) < tol)
+        {
+          alpha(j) = alpha_j_old;
+          continue;
+        }
+
+        // Determine value for alpha i using (16).
+        alpha(i) = alpha(i) + trainLabels(i) * trainLabels(j)
+                   * (alpha_j_old - alpha(j));
+
+        // Compute b1 and b2 using (17) and (18) respectively.
+        double b1 = intercept - E(i)
+                    - trainLabels(i) * (alpha(i) - alpha_i_old) *  K(i, j)
+                    - trainLabels(j) * (alpha(j) - alpha_j_old) *  K(i, j);
+        double b2 = intercept - E(j)
+                    - trainLabels(i) * (alpha(i) - alpha_i_old) *  K(i, j)
+                    - trainLabels(j) * (alpha(j) - alpha_j_old) *  K(j, j);
+
+        // Compute b by (19).
+        if (0 < alpha(i) && alpha(i) < regularization)
+          intercept = b1;
+        else if (0 < alpha(j) && alpha(j) < regularization)
+          intercept = b2;
+        else
+          intercept = (b1 + b2) / 2;
+
+        num_changed_alphas = num_changed_alphas + 1;
+      }
+    }
+
+    if (num_changed_alphas == 0)
+      count = count + 1;
+    else
+      count = 0;
+  }
+  // Calculating paramter values for linear kernel.
+  parameters = (data * (alpha.t() % trainLabels).t()).t();
+
+  trainingData = data;
+}
+
+template <typename MatType, typename KernelType>
+void KernelSVM<MatType, KernelType>::Classify(
+    const MatType& data,
+    arma::Row<size_t>& labels) const
+{
+  arma::mat scores;
+  Classify(data, labels, scores);
+}
+
+template <typename MatType, typename KernelType>
+void KernelSVM<MatType, KernelType>::Classify(
+    const MatType& data,
+    arma::Row<size_t>& labels,
+    arma::mat& scores) const
+{
+  Classify(data, scores);
+  double threshold = arma::as_scalar(arma::mean(scores, 1));
+
+  // Prepare necessary data.
+  labels.zeros(data.n_cols);
+
+  for (size_t i = 0; i< scores.n_elem; i++)
+  {
+    if (scores(i) >= threshold)
+      labels(i) = 1;
+    if (scores(i) < threshold)
+      labels(i) = 0;
+  }
+}
+
+template <typename MatType, typename KernelType>
+void KernelSVM<MatType, KernelType>::Classify(
+    const MatType& data,
+    arma::mat& scores) const
+{
+  if (fitIntercept)
+  {
+    // Giving prediction when non-linear kernel is used.
```

Comment:
We could use `fitIntercept` with linear kernel as well right?

Reply:
Just replaced `fitintercept` with `fitKernel` actually I am using it for different purpose.

Reply:
I don't think it makes a lot of sense for the user to specify an option such as `fitKernel`. Can you just remove that option and have `fitIntercept` instead?
The linear kernel should work similar to any other kernel.